solver/jobs.go

```diff
@@ -342,6 +342,14 @@ func (jl *Solver) loadUnlocked(v, parent Vertex, j *Job, cache map[Vertex]Vertex
 	// if same vertex is already loaded without cache just use that
 	st, ok := jl.actives[dgstWithoutCache]
 
+	if ok {
+		v = &vertexWithCacheOptions{
```

Comment:
Is this equivalent to `v = st.vtx` ?

If yes, then set a comment that we need we need to set `v` because it is added to the cache map.

Reply:
Probably? I'm not super familiar with the lifespan of `v` compared to `st.vtx`, but at least from a smoke test BuildKit didn't die with that change. 🙂 

I've pushed this change and added a comment 👍